### PR TITLE
Verify documentation

### PR DESCRIPTION
**Pull request by pynchmeister**
## Description

* Added documentation for contract verification at ```/explorer/apps/verify/README.md```.

* Added smart contract verification functionality to ```@modularcloud/sdk```

### CODE REVIEW

File: packages/@modularcloud/sdk/index.ts [hunk 1]
```diff
@@ -420,8 +430,33 @@ export function createModularCloud(baseUrl?: string): ModularCloud {
...
+        const response = await fetch(
+          `${baseUrl}/${normalizeNetworkId(networkId)}/contract-verification/source/${address}`
+        );
+
+        if (!response.ok) {
+          throw new Error("Failed to fetch verified contract source");
+        }
+
+        const json = (await response.json()) as APIResponse;
+        return VerificationResponseSchema.parse(json.result);
```
Comment (lzrscg) [inline packages/@modularcloud/sdk/index.ts:456]: This result should not be a `VerificationResponse` (`{ verified: boolean }`). It should contain a the text of the source code from the files returned by the API.

File: packages/@modularcloud/sdk/index.ts [hunk 2]
```diff
@@ -420,8 +430,33 @@ export function createModularCloud(baseUrl?: string): ModularCloud {
         const json = (await response.json()) as APIResponse;
         return CollectionResponseSchema.parse(json.result);
       },
+      isContractVerified: async (networkId: string, address: string) => {
+        const response = await fetch(
+          `${baseUrl}/api/contract-verification/fetch-verified?contractaddress=${address}`
```
Comment (lzrscg) [inline packages/@modularcloud/sdk/index.ts:435]: this base URL is different from the other URLs in this file, our API is located at the URL:
[contract-verification.vercel.app](https://contract-verification.vercel.app/)

Go ahead and hardcode this in

File: apps/verify/README.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+Query parameters:
+
+* contractaddress: The Ethereum address of the contract whose verification status you want to fetch.
+* readfiles(Optional):  True or False If you want to get the data of uploadfiles as string added along with the contract verification data
+
+  
+
+Responses:
+
+* 200 OK: The operation was successful. The response body will contain the verification data for the contract. If no contract is found with the provided address, { "isVerified": false } will be returned.
```
Comment (lzrscg) [inline apps/verify/README.md:172]: what is the schema if `readfiles` is `true`?

**Review by lzrscg — CHANGES_REQUESTED** [review]

File: apps/verify/README.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+### Contract Verification
+
+  
+
+This documentation provides details on the two key API endpoints included within the contract verification system. These endpoints allow for the verification and retrieval of smart contracts, providing developers a way to check and validate their contracts.
+
+  
+
+**/api/contract-verification/**
```
Comment (mhstom) [inline apps/verify/README.md:97]: maybe I am missing something but shouldn't this be `api/contract-verification/verify-contract` ?

File: apps/verify/README.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+Request method:
+
+POST
+
+  
+
+Description:
+
+This endpoint is used to verify a smart contract. The verification is done by making an API call to a Sourcify service (or a localhost if no Sourcify URL is provided in the environment variables). Depending on the response from this service, a record is created in the database that stores the verification status of the contract.
```
Comment (mhstom) [inline apps/verify/README.md:109]: the description looks more like https://github.com/modularcloud/explorer/blob/main/apps/verify/pages/verifyAndUpload.tsx to me.

the actual api doesn't seem to be doing that much? https://github.com/modularcloud/explorer/blob/main/apps/verify/pages/api/contract-verification/verify-contract.ts

File: apps/verify/README.md [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+
+[API routes](https://nextjs.org/docs/api-routes/introduction) can be accessed on [http://localhost:3000/api/hello](http://localhost:3000/api/hello). This endpoint can be edited in `pages/api/hello.js`.
+
+  
+
+The `pages/api` directory is mapped to `/api/*`. Files in this directory are treated as [API routes](https://nextjs.org/docs/api-routes/introduction) instead of React pages.
+
+  
+
+## Prisma
```
Comment (mhstom) [inline apps/verify/README.md:33]: where is the command to generate the db and tables?

File: apps/verify/README.md [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+
+Responses:
+
+* 200 OK: The contract was verified successfully. The response body will contain data from the Sourcify service.
+
+* 500 Internal Server Error: If the chainId isn't available for verification, or if there's an error while creating a record in the database, a 500 error will be returned along with an error message.
+
+  
+
+**/api/fetch-verified**
```
Comment (mhstom) [inline apps/verify/README.md:147]: and shouldn't this be /api/contract-verification/fetch-verified based on the repo route here? https://github.com/modularcloud/explorer/blob/main/apps/verify/pages/api/contract-verification/fetch-verified.ts

File: apps/verify/README.md (hunk 4 above)
Comment (pynchmeister) [reply apps/verify/README.md:97]: No, you are correct--I made a mistake. Fixed.

File: apps/verify/README.md (hunk 5 above)
Comment (pynchmeister) [reply apps/verify/README.md:109]: Ok, I've edited the description. Let me know if your point still stands.

File: apps/verify/README.md (hunk 6 above)
Comment (pynchmeister) [reply apps/verify/README.md:33]: added

File: apps/verify/README.md (hunk 7 above)
Comment (pynchmeister) [reply apps/verify/README.md]: This is correct, just changed it.

File: apps/verify/README.md (hunk 3 above)
Comment (pynchmeister) [reply apps/verify/README.md:186]:
```{
  "id": "string",
  "createdAt": "datetime",
  "updatedAt": "datetime",
  "verificationStatus": "string",
  "contractAddress": "string",
  "chainID": "string",
  "isVerified": "boolean",
  "uploadedUrl": "string",
  "files": "Object"
}
```

Added this to the docs

File: apps/verify/README.md [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+
+Request method:
+
+POST
+
+  
+
+Description:
+
+This API endpoint is designed to handle the verification of smart contracts. The verification is done by making an API call to a Sourcify service (or a localhost if no Sourcify URL is provided in the environment variables). Depending on the response from this service, a record is created in the database that stores the verification status of the contract.
```
Comment (mhstom) [inline apps/verify/README.md:115]: **After** the verification is done by making an API call to a Sourcify service, (or a localhost if no Sourcify URL is provided in the environment variables), this api create a record in the database that stores the verification status of the contract.

File: packages/@modularcloud/sdk/index.ts (hunk 1 above)
Comment (pynchmeister) [reply packages/@modularcloud/sdk/index.ts:456]: Noted.

File: apps/verify/README.md (hunk 8 above)
Comment (pynchmeister) [reply apps/verify/README.md:115]: I've replaced my description with yours.

File: apps/verify/README.md (hunk 3 above)
Comment (Aswanth-c) [reply apps/verify/README.md:200]: Schema with data for more convenience 
`{
    "id": 14,
    "contractAddress": "0x90cd9b9f69d1db3f66dd209784c90b92b0157b40",
    "chainID": "91002",
    "isVerified": true,
    "createdAt": "2023-06-21T21:11:39.028Z",
    "updatedAt": "2023-06-21T21:11:39.028Z",
    "verificationStatus": "FULL",
    "uploadedUrl": "https://verified-contracts.s3.us-west-2.amazonaws.com/0x90cd9b9f69d1db3f66dd209784c90b92b0157b40/0x90cd9b9f69d1db3f66dd209784c90b92b0157b40_sourcefiles.zip",
    "files": {
        "Caldera.sol": "// SPDX-License-Identifier: MIT\npragma solidity ^0.8.17;\n\ncontract HelloWorld {\n\n    string saySomething;\n\n    constructor() {\n        saySomething = \"Hello World!\";\n    }\n\n    function speak() public view returns(string memory) {\n        return saySomething;\n    }\n}",
        "metadata.json": "{\n\t\"compiler\": {\n\t\t\"version\": \"0.8.18+commit.87f61d96\"\n\t},\n\t\"language\": \"Solidity\",\n\t\"output\": {\n\t\t\"abi\": [\n\t\t\t{\n\t\t\t\t\"inputs\": [],\n\t\t\t\t\"stateMutability\": \"nonpayable\",\n\t\t\t\t\"type\": \"constructor\"\n\t\t\t},\n\t\t\t{\n\t\t\t\t\"inputs\": [],\n\t\t\t\t\"name\": \"speak\",\n\t\t\t\t\"outputs\": [\n\t\t\t\t\t{\n\t\t\t\t\t\t\"internalType\": \"string\",\n\t\t\t\t\t\t\"name\": \"\",\n\t\t\t\t\t\t\"type\": \"string\"\n\t\t\t\t\t}\n\t\t\t\t],\n\t\t\t\t\"stateMutability\": \"view\",\n\t\t\t\t\"type\": \"function\"\n\t\t\t}\n\t\t],\n\t\t\"devdoc\": {\n\t\t\t\"kind\": \"dev\",\n\t\t\t\"methods\": {},\n\t\t\t\"version\": 1\n\t\t},\n\t\t\"userdoc\": {\n\t\t\t\"kind\": \"user\",\n\t\t\t\"methods\": {},\n\t\t\t\"version\": 1\n\t\t}\n\t},\n\t\"settings\": {\n\t\t\"compilationTarget\": {\n\t\t\t\"Caldera.sol\": \"HelloWorld\"\n\t\t},\n\t\t\"evmVersion\": \"paris\",\n\t\t\"libraries\": {},\n\t\t\"metadata\": {\n\t\t\t\"bytecodeHash\": \"ipfs\"\n\t\t},\n\t\t\"optimizer\": {\n\t\t\t\"enabled\": false,\n\t\t\t\"runs\": 200\n\t\t},\n\t\t\"remappings\": []\n\t},\n\t\"sources\": {\n\t\t\"Caldera.sol\": {\n\t\t\t\"keccak256\": \"0x8f54065335b9a9019f30a20b9e2cb182aeec5d823ded5c67e71c22f42422b892\",\n\t\t\t\"license\": \"MIT\",\n\t\t\t\"urls\": [\n\t\t\t\t\"bzz-raw://21b5e98caff1f4831c23977aa77c9063fe735d843cb722ebda428715f586dcf2\",\n\t\t\t\t\"dweb:/ipfs/QmSk3zRdU2fxWEaw7em9PJ6LJuupxWgM1snGwqbNZgBB2W\"\n\t\t\t]\n\t\t}\n\t},\n\t\"version\": 1\n}"
    }
}`

File: packages/@modularcloud/sdk/schemas.ts [hunk 9]
```diff
@@ -133,3 +133,9 @@ export const CollectionResponseSchema = z.object({
   nextToken: z.string().optional(),
 });
 export type CollectionResponse = z.infer<typeof CollectionResponseSchema>;
+
+export const VerificationResponseSchema = z.object({
+  sourceCode: z.string(),
```
Comment (lzrscg) [inline packages/@modularcloud/sdk/schemas.ts:138]: why is this response just a single string. Isn't it possible to have multiple source files? what about file names?

File: packages/@modularcloud/sdk/index.ts [hunk 10]
```diff
@@ -420,8 +430,33 @@ export function createModularCloud(baseUrl?: string): ModularCloud {
...
+        const response = await fetch(
+          `https://contract-verification.vercel.app/api/contract-verification/fetch-verified?contractaddress=${address}`
+        );
+
+        if (!response.ok) {
+          throw new Error("Failed to fetch verified contract");
+        }
+
+        const json = (await response.json()) as APIResponse;
+        return VerificationResponseSchema.parse(json.result);
```
Comment (lzrscg) [inline packages/@modularcloud/sdk/index.ts:443]: I think this is the wrong schema. Can you test this as well?

**Review by lzrscg — CHANGES_REQUESTED** [review]

**Pull request merged**